NodeJS/__unit__/AgentClientMockito.test.ts

```diff
@@ -0,0 +1,42 @@
+import { AgentClient } from "../src/AgentClient";
+import { RestClient } from "../src/RestClient";
+import { mock, when, anyString, reset, instance, verify } from "ts-mockito";
+
+class AgentClientProtected extends AgentClient{
+    public fetchDefaultSkillNumber(){
+        return super.fetchDefaultSkillNumber()
+    }
+}
+
+describe("AgentClient.ts tests with mocked RestClient", () => {
+    let agentClient: AgentClientProtected;
+    let restClientMock: RestClient = mock(RestClient)
+    let restClientInstance: RestClient
+
+    beforeEach(() => {
+        reset(restClientMock)
+        restClientInstance = instance(restClientMock)
+
+    });
+    test("getDefaultSkillId should return default skill id", async () => {
+        let data = { "skillResponses": { '1': [{ name: 'DEFAULT_SKILL', number: 100 }] } }
+        let response = { data }
+        // set up mock
+        when(restClientMock.getSubAccountAgentSkills(anyString())).thenResolve((response))
+        // pass instance
+        agentClient = new AgentClientProtected("1", restClientInstance)
+        let skillId = await agentClient.fetchDefaultSkillNumber()
+        expect(skillId).toEqual(100)
+    })
+    test("getDefaultSkillId should return undefined when default skill is not returned", async () => {
```

Comment:
I like it, but I'd word it just a little differently as I wasn't sure what this meant when I first read it and had to look up `fetchDefaultSkillNumber` to understand what was happening.

`"getDefaultSkillId returns the default skill if it exists, else it will return 'undefined' to signify that it does not exist"`

Reply:
The wording is fine.  It is one test case. What you said sounds like two test cases.